deliver/lib/deliver/upload_metadata.rb

```diff
@@ -89,7 +89,7 @@ def upload(options)
       enabled_languages = detect_languages(options)
 
       app_store_version_localizations = verify_available_version_languages!(options, app, enabled_languages) unless options[:edit_live]
-      app_info = fetch_edit_app_info(app)
+      app_info = fetch_edit_app_info(app, max_retries: options[:version_check_wait_retry_limit])
```

Comment:
I would consider creating an object called RetryConfig and move all fields related to the retry (`max_retries`, `initial_wait_time)` into it.

This would change the `retry_if_nil` method  like this
```
    def retry_if_nil(message, retry_config:)
      wait_time = retry_config[:initial_wait_time]
      tries = retry_config[:tries]
```
and all other calls would pass a single parameter that could be initialized in one place.

Does that make sense?

Reply:
Yeah it does, just not sure if it's worth unifying these into an object, since it's just 2 configs? I usually do these type of things once there are 3 or more parameters 👀 Do you think this would make a significant difference? In terms of e.g. UX/DX, testability, maintainability…? @lacostej

Reply:
We fetch the  `options[:version_check_wait_retry_limit]` 5 times in the code, I suspect there should be a way to make this a bit better.

Reply:
Here's a proposal

1. we do not add any retry related parameter to the `fetch_*` methods. We even remove the `wait_time` one
2. we also remove the parameters from `retry_if_nil`, and instead, initialize the `initial_wait_time` from `options` within the implementation

this should remove most retry specific information in most of the code

3. remains the question of the tests. I guess we pass a default value of `0.01` to ensure that they do not take too long. There we could instead stub the `sleep` call instead, and just ensure it gets called the right amount of times. It will even make the tests slightly faster. [Stubbing sleep requires a bit more work to do right](https://stackoverflow.com/questions/1168845/rspec-stubbing-kernelsleep).

Reply:
I think this would be a good approach, I'll try to come back to this later this week 🙏 

If you have a clear understanding on how to implement this and wants to take a stab before I have time to get to it, that'd probably help us getting this out of the door faster as I don't think I'll have much availability in the next few days 🙇 if not, all good too! 🙏

Reply:
Hey team, I won't have time to look into this in the upcoming future 😢 @lacostej or anyone else, feel free to pick this up 🙏I'm also ok with merging this in as is and creating a separate task to refactor later (although I suspect it won't be picked up because it won't be a high priority 😞)

Reply:
Hello 👋 I will take it over and give it a shoot 😊

Reply:
@mollyIV 

Yes this is good. If you look at the [overall combined PR](https://github.com/fastlane/fastlane/compare/master...upload-metadata-options-init), the code is I think cleaner.

thanks for spending the time to do this!